src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,126 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int INDEX_NULL_KEY = 0;
+    private Node<K, V>[] table;
+    private int size;
+    private int modCount;
+    private int capacity;
+
+    public MyHashMap() {
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        modCount = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        table = (Node<K, V>[]) new Node[capacity];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        putVal(key, value, table);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        return findValue(key);
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private V findValue(K key) {
+        if (key == null) {
+            return table[INDEX_NULL_KEY].value;
+        }
+        V value = null;
+        for (int i = 1; i < capacity; i++) {
+            for (Node<K, V> current = table[i]; current != null; current = current.next) {
+                if (current.key.equals(key)) {
+                    value = current.value;
+                }
+            }
+        }
+        return value;
+    }
+
+    private void putVal(K key, V value, Node<K, V>[] node) {
+        if (isNeedResize()) {
+            resizeTable();
+            putAfter(key, value, table);
+        } else if (key == null) {
+            if (node[INDEX_NULL_KEY] == null) {
+                size++;
+            }
+            node[INDEX_NULL_KEY] = new Node<K, V>(key, value, null);
+        } else {
+            putAfter(key, value, node);
+        }
+    }
+
+    private void resizeTable() {
+        int oldCapacity = capacity;
+        capacity *= 2;
+        modCount = (int) (capacity * DEFAULT_LOAD_FACTOR);
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[capacity];
+        int oldSize = size;
+        for (int i = 0; i < oldCapacity; i++) {
+            for (Node<K, V> current = table[i]; current != null; current = current.next) {
+                putVal(current.key, current.value, newTable);
+            }
+        }
+        this.table = newTable;
```

Comment:
You can make all updates on the main table, but not on the new one. So will not need to pass the table as an argument.

Reply:
Really =)